runtime/directives/sky-app-link-external.directive.ts

```diff
@@ -0,0 +1,36 @@
+import { Directive, Input, HostListener, OnInit } from '@angular/core';
+import { SkyAppWindowRef } from '../window-ref';
+import { SkyAppConfig } from '@blackbaud/skyux-builder/runtime';
+import { HttpParams } from '@angular/common/http';
+
+@Directive({
+  selector: '[skyAppLinkExternal]'
+})
+export class SkyAppLinkExternalDirective implements OnInit {
+  @Input() private skyAppLinkExternal: string;
+  private externalSPAUrl: string;
+
+  constructor(
+    private skyAppConfig: SkyAppConfig,
+    private window: SkyAppWindowRef
+  ) { }
+
+  public ngOnInit() {
+    const queryParams: any = this.skyAppConfig.runtime.params.getAll();
+    const urlParts = this.skyAppLinkExternal.split('?');
```

Comment:
This assumes the `skyAppLinkExternal` value is a string, but Angular's `routerLink` (and therefore `skyAppLink`) also allow arrays so that URLs can be built up in parts that are dynamic for consumer convenience.  For instance, this is valid where `part2` is a public property of the component:

`<a [skyAppLink]="['part-1', part2, 'part-3']">Dynamic link</a>`